src/HealthChecks.AzureServiceBus/AzureServiceBusHealthCheck.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.Collections.Concurrent;
+using Azure.Core;
+using Azure.Messaging.ServiceBus.Administration;
+
+namespace HealthChecks.AzureServiceBus
+{
+    public abstract class AzureServiceBusHealthCheck
+    {
+        protected static readonly ConcurrentDictionary<string, ServiceBusAdministrationClient>
+            ManagementClientConnections = new();
+
+        private string ConnectionString { get; }
+
+        protected string Prefix => ConnectionString ?? Endpoint;
+
+        private string Endpoint { get; }
+        private TokenCredential TokenCredential { get; }
+
+
+        protected AzureServiceBusHealthCheck(string connectionString)
+        {
+            if (string.IsNullOrEmpty(connectionString))
+            {
+                throw new ArgumentNullException(nameof(connectionString));
+            }
+
+            ConnectionString = connectionString;
+        }
+
+        protected AzureServiceBusHealthCheck(string endpoint, TokenCredential tokenCredential)
+        {
+            if (string.IsNullOrEmpty(endpoint))
+            {
+                throw new ArgumentNullException(nameof(endpoint));
+            }
+
+            if (tokenCredential == null)
+            {
+                throw new ArgumentNullException(nameof(tokenCredential));
+            }
+
+            Endpoint = endpoint;
+            TokenCredential = tokenCredential;
+        }
+
+
+        protected ServiceBusAdministrationClient CreateManagementClient()
+        {
+            ServiceBusAdministrationClient managementClient;
+            if (TokenCredential != null)
```

Comment:
stylistic but this could be inlined using `?` and `:`

```c#
protected ServiceBusAdministrationClient CreateManagementClient() =>
  TokenCredential != null
  ? new ServiceBusAdministrationClient(Endpoint, TokenCredential)
  : new ServiceBusAdministrationClient(ConnectionString);
```